installer/installer.py

```diff
@@ -0,0 +1,600 @@
+                     DEFAULT_CHEQD_USER,
+                     DEFAULT_CHEQD_USER)
+
+        # Setting up the external_address
+        if self.interviewer.external_address:
+            self.exec(f"sudo -u {DEFAULT_CHEQD_USER} cheqd-noded configure p2p external-address {self.interviewer.external_address}")
+
+        # Setting up the seeds
+        seeds = self.exec(f"curl -s {SEEDS_URL_TEMPLATE.format(self.interviewer.chain)}").stdout.decode("utf-8").strip()
+        self.exec(f"sudo -u {DEFAULT_CHEQD_USER} cheqd-noded configure p2p seeds {seeds}")
+
+    def prepare_cheqd_user(self):
+        self.log(f"Create group, {DEFAULT_CHEQD_USER} by default")
+        self.exec(f"addgroup {DEFAULT_CHEQD_USER} --quiet")
+
+        self.log(f"Create user, {DEFAULT_CHEQD_USER} by default")
+        self.exec(
+            f"adduser --system {DEFAULT_CHEQD_USER} --home {self.interviewer.home_dir} --shell /bin/bash --ingroup {DEFAULT_CHEQD_USER} --quiet")
+
+    def mkdir_p(self, dir_name):
+        try:
+            os.mkdir(dir_name)
+        except FileExistsError as err:
+            self.log(f"Directory {dir_name} already exists")
+
+    def setup_log_dir(self):
+        self.mkdir_p(self.cheqd_log_dir)
+        Path(os.path.join(self.cheqd_log_dir, "stdout.log")).touch()
+        self.exec(f"chown -R syslog:syslog {self.cheqd_log_dir}")
+
+    def setup_cosmovisor(self):
+        self.exec(f"wget -qO - {DEFAULT_COSMOVISOR_TAR_URL}  | tar xz")
+        self.mkdir_p(self.cosmovisor_root_dir)
+        self.mkdir_p(os.path.join(self.cosmovisor_root_dir, "genesis"))
+        self.mkdir_p(os.path.join(self.cosmovisor_root_dir, "genesis/bin"))
+        self.mkdir_p(os.path.join(self.cosmovisor_root_dir, "upgrades"))
+        if not os.path.exists(os.path.join(DEFAULT_INSTALL_PATH, "cosmovisor")):
+            self.log(f"Moving cosmovisor binary to default installation directory")
+            shutil.move("./cosmovisor", DEFAULT_INSTALL_PATH)
+
+        self.log(f"Moving binary from {self.binary_path} to {self.cosmovisor_bin_path}")
+        self.exec("sudo mv {} {}".format(self.binary_path, self.cosmovisor_bin_path))
+
+        if not os.path.exists(os.path.join(DEFAULT_INSTALL_PATH, DEFAULT_BINARY_NAME)):
+            self.log(f"Making symlink to {self.cosmovisor_bin_path}")
+            os.symlink(self.cosmovisor_bin_path,
+                       os.path.join(DEFAULT_INSTALL_PATH, DEFAULT_BINARY_NAME))
+
+        self.log(f"Changing owner to {DEFAULT_CHEQD_USER} user")
+        self.exec(f"chown -R {DEFAULT_CHEQD_USER}:{DEFAULT_CHEQD_USER} {self.cosmovisor_root_dir}")
+
+    def untar_from_snapshot(self):
+        self.mkdir_p(self.cheqd_data_dir)
+        cmd = f"wget -qO - {self.interviewer.snapshot_url}  | sudo -u {DEFAULT_CHEQD_USER} tar xzf - -C {os.path.join(self.cheqd_root_dir, 'data')}"
+        thread = threading.Thread(target=functools.partial(self.exec, cmd))
+        thread.start()
+        sec_counter = 0
+        while thread.is_alive():
```

Comment:
What will happen in case of error in the download thread?

Reply:
I checked such situation. It will exit with error:
```
Reason of failure: Command 'wget -qO - https://cheqd-node-backups.ams3.cdn.digitaloceanspaces.com/testnet/latest/cheqd-testnet-4_2022-06-21.tar.gz1234  | sudo -u cheqd tar xzf - -C /home/cheqd/.cheqdnode/data' returned non-zero exit status 127.
Exiting....
```